pilot/pkg/serviceregistry/kube/controller/ambient/policies.go

```diff
@@ -207,26 +207,31 @@ func PolicyCollections(
 		PeerAuthDerivedPolicies,
 		DefaultPolicy.AsCollection(),
 		ImplicitWaypointPolicies,
-	}, krt.WithName("Policies"))
+	}, opts.WithName("Policies")...)
 	return AuthzDerivedPolicies, Policies
 }
 
-func implicitWaypointPolicyName(waypoint *Waypoint) string {
-	if !features.DefaultAllowFromWaypoint || waypoint == nil || len(waypoint.ServiceAccounts) == 0 {
+func implicitWaypointPolicyName(flags FeatureFlags, waypoint *Waypoint) string {
```

Comment:
prefer a global var rather than passing it into every function

Reply:
The reason this is done is to avoid a dependency on **global mutable** state, which triggers race conditions. This is discussed a bit more in the PR description

Reply:
not in the test code which mutates them for specific tests